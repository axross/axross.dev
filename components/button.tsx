import { css, cx } from "@linaria/core";
import * as React from "react";
import type { IconProps } from "react-feather";

export enum ButtonVariant {
  default = "default",
  inverted = "inverted",
}

export enum ButtonSize {
  sm = "sm",
  md = "md",
  lg = "lg",
  xl = "xl",
}

export interface ButtonProps extends React.HTMLAttributes<HTMLButtonElement> {
  variant?: ButtonVariant;
  size?: ButtonSize;
  icon?: React.ReactElement<IconProps>;
  className?: string;
  style?: React.CSSProperties;
}

export const Button: React.FC<ButtonProps> = ({
  variant = ButtonVariant.default,
  size = ButtonSize.md,
  icon,
  className,
  children,
  ...props
}) => {
  return (
    <button
      className={cx(
        css`
          appearance: none;
          display: flex;
          align-items: center;
          justify-content: center;
          max-width: 100%;
          border: none;
          box-shadow: transparent 0 0 0 4px;
          color: var(--color-fg-strong);
          cursor: pointer;
          transition: background-color 150ms ease-in-out,
            box-shadow 300ms ease-in-out, color 300ms ease-in-out;

          &:focus {
            box-shadow: var(--color-fg-blue-weak) 0 0 0 4px;
            outline: none;
          }
        `,
        COLORS[variant],
        SIZES[size],
        className
      )}
      {...props}
    >
      {icon
        ? React.cloneElement(icon, {
            ...icon.props,
            className: cx(
              css`
                color: currentColor;
              `,
              ICON_SIZES[size],
              children
                ? css`
                    margin-inline-end: var(--space-sm);
                  `
                : null,
              icon.props.className
            ),
          })
        : null}

      <span
        className={cx(
          css`
            display: -webkit-box;
            -webkit-box-orient: vertical;
            -webkit-line-clamp: 1;
            overflow: hidden;
          `,
          TEXT_SIZES[size]
        )}
      >
        {children}
      </span>
    </button>
  );
};

const COLORS: Record<ButtonVariant, string> = {
  [ButtonVariant.default]: css`
    background-color: var(--color-bg-input);
    color: var(--color-fg-strong);

    &:hover,
    &:active {
      background-color: var(--color-bg-input-active);
    }
  `,
  [ButtonVariant.inverted]: css`
    background-color: var(--color-fg);
    color: var(--color-bg);

    &:hover,
    &:active {
      background-color: var(--color-fg-strong);
    }
  `,
};

const SIZES: Record<ButtonSize, string> = {
  [ButtonSize.sm]: css`
    height: 32px;
    padding-inline-start: var(--space-md);
    padding-inline-end: var(--space-md);
    border-radius: 8px;
  `,
  [ButtonSize.md]: css`
    height: 40px;
    padding-inline-start: var(--space-md);
    padding-inline-end: var(--space-md);
    border-radius: 8px;
  `,
  [ButtonSize.lg]: css`
    height: 48px;
    padding-inline-start: var(--space-lg);
    padding-inline-end: var(--space-lg);
    border-radius: 12px;
  `,
  [ButtonSize.xl]: css`
    height: 64px;
    padding-inline-start: var(--space-md);
    padding-inline-end: var(--space-md);
    border-radius: 16px;
  `,
};

const ICON_SIZES: Record<ButtonSize, string> = {
  [ButtonSize.sm]: css`
    width: 16px;
    height: 16px;
  `,
  [ButtonSize.md]: css`
    width: 20px;
    height: 20px;
  `,
  [ButtonSize.lg]: css`
    width: 20px;
    height: 20px;
  `,
  [ButtonSize.xl]: css`
    width: 32px;
    height: 32px;
  `,
};

const TEXT_SIZES: Record<ButtonSize, string> = {
  [ButtonSize.sm]: css`
    font-size: var(--font-size-sm);
  `,
  [ButtonSize.md]: css`
    font-size: var(--font-size-md);
  `,
  [ButtonSize.lg]: css`
    font-size: var(--font-size-md);
  `,
  [ButtonSize.xl]: css`
    font-size: var(--font-size-xxl);
  `,
};
