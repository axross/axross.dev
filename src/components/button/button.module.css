@layer components {
  .root {
    appearance: none;
    display: inline-flex;
    align-items: center;
    justify-content: center;
    min-width: var(--__button-min-width);
    height: var(--__button-height);
    padding-inline: var(--__button-padding-inline);
    border-radius: var(--radius-sm);
    outline: none;
    overflow: hidden;
    cursor: pointer;
    user-select: none;
    transition:
      background-color var(--ease-in-out) var(--duration-md),
      border var(--ease-in-out) var(--duration-md);

    /* invalid data-variant will fallback to data-variant="solid" */
    background-color: var(--__button-solid-bg-color);
    border: none;
    color: var(--__button-solid-fg-color);
    font-size: var(--__button-font-size);

    /* invalid data-intent will fallback to data-intent="neutral" */
    --__button-fg-color: var(--neutral-12);
    --__button-solid-bg-color: var(--neutral-12);
    --__button-solid-fg-color: var(--neutral-1);
    --__button-ghost-bg-color: var(--neutral-a3);
    --__button-border-color: var(--neutral-6);
    --__button-outline-color: var(--action-8);

    /* invalid data-size will fallback to data-size="md" */
    --__button-min-width: var(--size-80);
    --__button-height: var(--size-40);
    --__button-padding-inline: var(--size-16);
    --__button-radius: var(--radius-md);
    --__button-font-size: var(--text-md);
  }

  .root:focus-visible {
    outline-width: var(--size-3);
    outline-offset: var(--size-3);
  }

  .root:active {
    transform: scale3d(95%, 95%, 95%);
  }

  .root[aria-disabled="true"],
  .root[aria-busy="true"] {
    pointer-events: none;
    cursor: not-allowed;
    opacity: 50%;
  }

  .root[data-variant="outline"] {
    background-color: transparent;
    border: var(--size-1) solid var(--__button-border-color);
    color: var(--__button-fg-color);
  }

  .root[data-variant="ghost"] {
    background-color: transparent;
    color: var(--__button-fg-color);
  }

  .root[data-variant="ghost"]:hover {
    background-color: var(--__button-ghost-bg-color);
  }

  .root[data-intent="danger"] {
    --__button-fg-color: var(--danger-10);
    --__button-solid-bg-color: var(--danger-10);
    --__button-solid-fg-color: var(--danger-1);
    --__button-ghost-bg-color: var(--danger-3);
    --__button-border-color: var(--danger-6);
    --__button-outline-color: var(--danger-7);

    @container style(--theme: dark) {
      --__button-solid-fg-color: var(--neutral-12);
    }
  }

  .root[data-size="sm"] {
    --__button-min-width: var(--size-48);
    --__button-height: var(--size-32);
    --__button-padding-inline: var(--size-12);
    --__button-radius: var(--radius-sm);
    --__button-font-size: var(--text-sm);
  }
}
