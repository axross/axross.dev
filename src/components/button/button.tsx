import { Slot } from "@radix-ui/react-slot";
import { clsx } from "clsx";
import {
  type ComponentPropsWithoutRef,
  type ElementRef,
  forwardRef,
} from "react";
import css from "./button.module.css";

type ButtonProps = ComponentPropsWithoutRef<"button"> & {
  readonly variant: "ghost" | "link" | "outline" | "solid";
  readonly intent: "danger" | "neutral";
  readonly size?: "md" | "sm";
  readonly asChild?: boolean;
};

const Button = forwardRef<ElementRef<"button">, ButtonProps>(
  (
    {
      className,
      variant,
      intent,
      size = "md",
      disabled = false,
      asChild = false,
      ...props
    },
    ref,
  ) => {
    const Component = asChild ? Slot : "button";

    return (
      <Component
        ref={ref}
        className={clsx(css.root, className)}
        aria-disabled={disabled ? true : undefined}
        data-variant={variant}
        data-intent={intent}
        data-size={size}
        {...props}
      />
    );
  },
);

Button.displayName = "Button";

export { Button };
