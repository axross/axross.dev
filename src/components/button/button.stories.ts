import { type Meta, type StoryObj } from "@storybook/react";
import { Button } from "./button";

const meta = {
  component: Button,
  argTypes: {
    intent: {
      control: "radio",
      options: ["action", "danger", "neutral"],
    },
    variant: {
      control: "radio",
      options: ["solid", "outline", "ghost"],
    },
    disabled: {
      control: "boolean",
    },
  },
  args: {
    children: "Hello, world!",
  },
} satisfies Meta<typeof Button>;

export default meta;

type Story = StoryObj<typeof meta>;

export const Solid: Story = {
  args: {
    variant: "solid",
    intent: "neutral",
  },
};

export const Outline: Story = {
  args: {
    ...Solid.args,
    variant: "outline",
  },
};

export const Ghost: Story = {
  args: {
    ...Solid.args,
    variant: "ghost",
  },
};

export const Small: Story = {
  args: {
    ...Solid.args,
    size: "sm",
  },
};
