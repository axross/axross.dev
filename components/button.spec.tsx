import { fireEvent, render } from "@testing-library/react";
import * as React from "react";
import { Mic } from "react-feather";
import { Button, ButtonSize, ButtonVariant } from "./button";

describe("<Button>", () => {
  it("matches the snapshot", () => {
    for (const variant of [ButtonVariant.default, ButtonVariant.inverted]) {
      for (const size of [
        ButtonSize.sm,
        ButtonSize.md,
        ButtonSize.lg,
        ButtonSize.xl,
      ]) {
        for (const hasIcon of [true, false]) {
          for (const hasChildren of [true, false]) {
            const { container } = render(
              <Button
                variant={variant}
                size={size}
                icon={hasIcon ? <Mic /> : undefined}
                data-testid="button"
              >
                {hasChildren ? "Lorem Ipsum" : undefined}
              </Button>
            );

            expect(container).toMatchSnapshot(
              `variant: ${variant}, size: ${size}, icon: ${hasIcon}, children: ${hasChildren}`
            );
          }
        }
      }
    }
  });

  it("calls props.onClick() when it's clicked", () => {
    const onClick = jest.fn();
    const { getByTestId } = render(
      <Button onClick={onClick} data-testid="button" />
    );

    fireEvent.click(getByTestId("button"));

    expect(onClick).toHaveBeenCalledTimes(1);
  });
});
