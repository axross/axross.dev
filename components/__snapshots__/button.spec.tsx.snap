// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button> matches the snapshot: variant: default, size: lg, icon: false, children: false 1`] = `
<div>
  <button
    class="b11jtqo5 b1gnzyb2 b1l9461r"
    data-testid="button"
  >
    <span
      class="slvi054 bo94lri"
    />
  </button>
</div>
`;

exports[`<Button> matches the snapshot: variant: default, size: lg, icon: false, children: true 1`] = `
<div>
  <button
    class="b11jtqo5 b1gnzyb2 b1l9461r"
    data-testid="button"
  >
    <span
      class="slvi054 bo94lri"
    >
      Lorem Ipsum
    </span>
  </button>
</div>
`;

exports[`<Button> matches the snapshot: variant: default, size: lg, icon: true, children: false 1`] = `
<div>
  <button
    class="b11jtqo5 b1gnzyb2 b1l9461r"
    data-testid="button"
  >
    <svg
      class="c108jgnp b4190x9"
      fill="none"
      height="24"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M12 1a3 3 0 0 0-3 3v8a3 3 0 0 0 6 0V4a3 3 0 0 0-3-3z"
      />
      <path
        d="M19 10v2a7 7 0 0 1-14 0v-2"
      />
      <line
        x1="12"
        x2="12"
        y1="19"
        y2="23"
      />
      <line
        x1="8"
        x2="16"
        y1="23"
        y2="23"
      />
    </svg>
    <span
      class="slvi054 bo94lri"
    />
  </button>
</div>
`;

exports[`<Button> matches the snapshot: variant: default, size: lg, icon: true, children: true 1`] = `
<div>
  <button
    class="b11jtqo5 b1gnzyb2 b1l9461r"
    data-testid="button"
  >
    <svg
      class="c108jgnp b4190x9 c75i3tf"
      fill="none"
      height="24"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M12 1a3 3 0 0 0-3 3v8a3 3 0 0 0 6 0V4a3 3 0 0 0-3-3z"
      />
      <path
        d="M19 10v2a7 7 0 0 1-14 0v-2"
      />
      <line
        x1="12"
        x2="12"
        y1="19"
        y2="23"
      />
      <line
        x1="8"
        x2="16"
        y1="23"
        y2="23"
      />
    </svg>
    <span
      class="slvi054 bo94lri"
    >
      Lorem Ipsum
    </span>
  </button>
</div>
`;

exports[`<Button> matches the snapshot: variant: default, size: md, icon: false, children: false 1`] = `
<div>
  <button
    class="b11jtqo5 b1gnzyb2 bq44b7t"
    data-testid="button"
  >
    <span
      class="slvi054 br7adeq"
    />
  </button>
</div>
`;

exports[`<Button> matches the snapshot: variant: default, size: md, icon: false, children: true 1`] = `
<div>
  <button
    class="b11jtqo5 b1gnzyb2 bq44b7t"
    data-testid="button"
  >
    <span
      class="slvi054 br7adeq"
    >
      Lorem Ipsum
    </span>
  </button>
</div>
`;

exports[`<Button> matches the snapshot: variant: default, size: md, icon: true, children: false 1`] = `
<div>
  <button
    class="b11jtqo5 b1gnzyb2 bq44b7t"
    data-testid="button"
  >
    <svg
      class="c108jgnp b1ig5l0y"
      fill="none"
      height="24"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M12 1a3 3 0 0 0-3 3v8a3 3 0 0 0 6 0V4a3 3 0 0 0-3-3z"
      />
      <path
        d="M19 10v2a7 7 0 0 1-14 0v-2"
      />
      <line
        x1="12"
        x2="12"
        y1="19"
        y2="23"
      />
      <line
        x1="8"
        x2="16"
        y1="23"
        y2="23"
      />
    </svg>
    <span
      class="slvi054 br7adeq"
    />
  </button>
</div>
`;

exports[`<Button> matches the snapshot: variant: default, size: md, icon: true, children: true 1`] = `
<div>
  <button
    class="b11jtqo5 b1gnzyb2 bq44b7t"
    data-testid="button"
  >
    <svg
      class="c108jgnp b1ig5l0y c75i3tf"
      fill="none"
      height="24"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M12 1a3 3 0 0 0-3 3v8a3 3 0 0 0 6 0V4a3 3 0 0 0-3-3z"
      />
      <path
        d="M19 10v2a7 7 0 0 1-14 0v-2"
      />
      <line
        x1="12"
        x2="12"
        y1="19"
        y2="23"
      />
      <line
        x1="8"
        x2="16"
        y1="23"
        y2="23"
      />
    </svg>
    <span
      class="slvi054 br7adeq"
    >
      Lorem Ipsum
    </span>
  </button>
</div>
`;

exports[`<Button> matches the snapshot: variant: default, size: sm, icon: false, children: false 1`] = `
<div>
  <button
    class="b11jtqo5 b1gnzyb2 b14eo67x"
    data-testid="button"
  >
    <span
      class="slvi054 bj3h2v"
    />
  </button>
</div>
`;

exports[`<Button> matches the snapshot: variant: default, size: sm, icon: false, children: true 1`] = `
<div>
  <button
    class="b11jtqo5 b1gnzyb2 b14eo67x"
    data-testid="button"
  >
    <span
      class="slvi054 bj3h2v"
    >
      Lorem Ipsum
    </span>
  </button>
</div>
`;

exports[`<Button> matches the snapshot: variant: default, size: sm, icon: true, children: false 1`] = `
<div>
  <button
    class="b11jtqo5 b1gnzyb2 b14eo67x"
    data-testid="button"
  >
    <svg
      class="c108jgnp b106izpi"
      fill="none"
      height="24"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M12 1a3 3 0 0 0-3 3v8a3 3 0 0 0 6 0V4a3 3 0 0 0-3-3z"
      />
      <path
        d="M19 10v2a7 7 0 0 1-14 0v-2"
      />
      <line
        x1="12"
        x2="12"
        y1="19"
        y2="23"
      />
      <line
        x1="8"
        x2="16"
        y1="23"
        y2="23"
      />
    </svg>
    <span
      class="slvi054 bj3h2v"
    />
  </button>
</div>
`;

exports[`<Button> matches the snapshot: variant: default, size: sm, icon: true, children: true 1`] = `
<div>
  <button
    class="b11jtqo5 b1gnzyb2 b14eo67x"
    data-testid="button"
  >
    <svg
      class="c108jgnp b106izpi c75i3tf"
      fill="none"
      height="24"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M12 1a3 3 0 0 0-3 3v8a3 3 0 0 0 6 0V4a3 3 0 0 0-3-3z"
      />
      <path
        d="M19 10v2a7 7 0 0 1-14 0v-2"
      />
      <line
        x1="12"
        x2="12"
        y1="19"
        y2="23"
      />
      <line
        x1="8"
        x2="16"
        y1="23"
        y2="23"
      />
    </svg>
    <span
      class="slvi054 bj3h2v"
    >
      Lorem Ipsum
    </span>
  </button>
</div>
`;

exports[`<Button> matches the snapshot: variant: default, size: xl, icon: false, children: false 1`] = `
<div>
  <button
    class="b11jtqo5 b1gnzyb2 b1vby7g1"
    data-testid="button"
  >
    <span
      class="slvi054 b276tsp"
    />
  </button>
</div>
`;

exports[`<Button> matches the snapshot: variant: default, size: xl, icon: false, children: true 1`] = `
<div>
  <button
    class="b11jtqo5 b1gnzyb2 b1vby7g1"
    data-testid="button"
  >
    <span
      class="slvi054 b276tsp"
    >
      Lorem Ipsum
    </span>
  </button>
</div>
`;

exports[`<Button> matches the snapshot: variant: default, size: xl, icon: true, children: false 1`] = `
<div>
  <button
    class="b11jtqo5 b1gnzyb2 b1vby7g1"
    data-testid="button"
  >
    <svg
      class="c108jgnp bkz0lf"
      fill="none"
      height="24"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M12 1a3 3 0 0 0-3 3v8a3 3 0 0 0 6 0V4a3 3 0 0 0-3-3z"
      />
      <path
        d="M19 10v2a7 7 0 0 1-14 0v-2"
      />
      <line
        x1="12"
        x2="12"
        y1="19"
        y2="23"
      />
      <line
        x1="8"
        x2="16"
        y1="23"
        y2="23"
      />
    </svg>
    <span
      class="slvi054 b276tsp"
    />
  </button>
</div>
`;

exports[`<Button> matches the snapshot: variant: default, size: xl, icon: true, children: true 1`] = `
<div>
  <button
    class="b11jtqo5 b1gnzyb2 b1vby7g1"
    data-testid="button"
  >
    <svg
      class="c108jgnp bkz0lf c75i3tf"
      fill="none"
      height="24"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M12 1a3 3 0 0 0-3 3v8a3 3 0 0 0 6 0V4a3 3 0 0 0-3-3z"
      />
      <path
        d="M19 10v2a7 7 0 0 1-14 0v-2"
      />
      <line
        x1="12"
        x2="12"
        y1="19"
        y2="23"
      />
      <line
        x1="8"
        x2="16"
        y1="23"
        y2="23"
      />
    </svg>
    <span
      class="slvi054 b276tsp"
    >
      Lorem Ipsum
    </span>
  </button>
</div>
`;

exports[`<Button> matches the snapshot: variant: inverted, size: lg, icon: false, children: false 1`] = `
<div>
  <button
    class="b11jtqo5 bdi40cx b1l9461r"
    data-testid="button"
  >
    <span
      class="slvi054 bo94lri"
    />
  </button>
</div>
`;

exports[`<Button> matches the snapshot: variant: inverted, size: lg, icon: false, children: true 1`] = `
<div>
  <button
    class="b11jtqo5 bdi40cx b1l9461r"
    data-testid="button"
  >
    <span
      class="slvi054 bo94lri"
    >
      Lorem Ipsum
    </span>
  </button>
</div>
`;

exports[`<Button> matches the snapshot: variant: inverted, size: lg, icon: true, children: false 1`] = `
<div>
  <button
    class="b11jtqo5 bdi40cx b1l9461r"
    data-testid="button"
  >
    <svg
      class="c108jgnp b4190x9"
      fill="none"
      height="24"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M12 1a3 3 0 0 0-3 3v8a3 3 0 0 0 6 0V4a3 3 0 0 0-3-3z"
      />
      <path
        d="M19 10v2a7 7 0 0 1-14 0v-2"
      />
      <line
        x1="12"
        x2="12"
        y1="19"
        y2="23"
      />
      <line
        x1="8"
        x2="16"
        y1="23"
        y2="23"
      />
    </svg>
    <span
      class="slvi054 bo94lri"
    />
  </button>
</div>
`;

exports[`<Button> matches the snapshot: variant: inverted, size: lg, icon: true, children: true 1`] = `
<div>
  <button
    class="b11jtqo5 bdi40cx b1l9461r"
    data-testid="button"
  >
    <svg
      class="c108jgnp b4190x9 c75i3tf"
      fill="none"
      height="24"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M12 1a3 3 0 0 0-3 3v8a3 3 0 0 0 6 0V4a3 3 0 0 0-3-3z"
      />
      <path
        d="M19 10v2a7 7 0 0 1-14 0v-2"
      />
      <line
        x1="12"
        x2="12"
        y1="19"
        y2="23"
      />
      <line
        x1="8"
        x2="16"
        y1="23"
        y2="23"
      />
    </svg>
    <span
      class="slvi054 bo94lri"
    >
      Lorem Ipsum
    </span>
  </button>
</div>
`;

exports[`<Button> matches the snapshot: variant: inverted, size: md, icon: false, children: false 1`] = `
<div>
  <button
    class="b11jtqo5 bdi40cx bq44b7t"
    data-testid="button"
  >
    <span
      class="slvi054 br7adeq"
    />
  </button>
</div>
`;

exports[`<Button> matches the snapshot: variant: inverted, size: md, icon: false, children: true 1`] = `
<div>
  <button
    class="b11jtqo5 bdi40cx bq44b7t"
    data-testid="button"
  >
    <span
      class="slvi054 br7adeq"
    >
      Lorem Ipsum
    </span>
  </button>
</div>
`;

exports[`<Button> matches the snapshot: variant: inverted, size: md, icon: true, children: false 1`] = `
<div>
  <button
    class="b11jtqo5 bdi40cx bq44b7t"
    data-testid="button"
  >
    <svg
      class="c108jgnp b1ig5l0y"
      fill="none"
      height="24"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M12 1a3 3 0 0 0-3 3v8a3 3 0 0 0 6 0V4a3 3 0 0 0-3-3z"
      />
      <path
        d="M19 10v2a7 7 0 0 1-14 0v-2"
      />
      <line
        x1="12"
        x2="12"
        y1="19"
        y2="23"
      />
      <line
        x1="8"
        x2="16"
        y1="23"
        y2="23"
      />
    </svg>
    <span
      class="slvi054 br7adeq"
    />
  </button>
</div>
`;

exports[`<Button> matches the snapshot: variant: inverted, size: md, icon: true, children: true 1`] = `
<div>
  <button
    class="b11jtqo5 bdi40cx bq44b7t"
    data-testid="button"
  >
    <svg
      class="c108jgnp b1ig5l0y c75i3tf"
      fill="none"
      height="24"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M12 1a3 3 0 0 0-3 3v8a3 3 0 0 0 6 0V4a3 3 0 0 0-3-3z"
      />
      <path
        d="M19 10v2a7 7 0 0 1-14 0v-2"
      />
      <line
        x1="12"
        x2="12"
        y1="19"
        y2="23"
      />
      <line
        x1="8"
        x2="16"
        y1="23"
        y2="23"
      />
    </svg>
    <span
      class="slvi054 br7adeq"
    >
      Lorem Ipsum
    </span>
  </button>
</div>
`;

exports[`<Button> matches the snapshot: variant: inverted, size: sm, icon: false, children: false 1`] = `
<div>
  <button
    class="b11jtqo5 bdi40cx b14eo67x"
    data-testid="button"
  >
    <span
      class="slvi054 bj3h2v"
    />
  </button>
</div>
`;

exports[`<Button> matches the snapshot: variant: inverted, size: sm, icon: false, children: true 1`] = `
<div>
  <button
    class="b11jtqo5 bdi40cx b14eo67x"
    data-testid="button"
  >
    <span
      class="slvi054 bj3h2v"
    >
      Lorem Ipsum
    </span>
  </button>
</div>
`;

exports[`<Button> matches the snapshot: variant: inverted, size: sm, icon: true, children: false 1`] = `
<div>
  <button
    class="b11jtqo5 bdi40cx b14eo67x"
    data-testid="button"
  >
    <svg
      class="c108jgnp b106izpi"
      fill="none"
      height="24"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M12 1a3 3 0 0 0-3 3v8a3 3 0 0 0 6 0V4a3 3 0 0 0-3-3z"
      />
      <path
        d="M19 10v2a7 7 0 0 1-14 0v-2"
      />
      <line
        x1="12"
        x2="12"
        y1="19"
        y2="23"
      />
      <line
        x1="8"
        x2="16"
        y1="23"
        y2="23"
      />
    </svg>
    <span
      class="slvi054 bj3h2v"
    />
  </button>
</div>
`;

exports[`<Button> matches the snapshot: variant: inverted, size: sm, icon: true, children: true 1`] = `
<div>
  <button
    class="b11jtqo5 bdi40cx b14eo67x"
    data-testid="button"
  >
    <svg
      class="c108jgnp b106izpi c75i3tf"
      fill="none"
      height="24"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M12 1a3 3 0 0 0-3 3v8a3 3 0 0 0 6 0V4a3 3 0 0 0-3-3z"
      />
      <path
        d="M19 10v2a7 7 0 0 1-14 0v-2"
      />
      <line
        x1="12"
        x2="12"
        y1="19"
        y2="23"
      />
      <line
        x1="8"
        x2="16"
        y1="23"
        y2="23"
      />
    </svg>
    <span
      class="slvi054 bj3h2v"
    >
      Lorem Ipsum
    </span>
  </button>
</div>
`;

exports[`<Button> matches the snapshot: variant: inverted, size: xl, icon: false, children: false 1`] = `
<div>
  <button
    class="b11jtqo5 bdi40cx b1vby7g1"
    data-testid="button"
  >
    <span
      class="slvi054 b276tsp"
    />
  </button>
</div>
`;

exports[`<Button> matches the snapshot: variant: inverted, size: xl, icon: false, children: true 1`] = `
<div>
  <button
    class="b11jtqo5 bdi40cx b1vby7g1"
    data-testid="button"
  >
    <span
      class="slvi054 b276tsp"
    >
      Lorem Ipsum
    </span>
  </button>
</div>
`;

exports[`<Button> matches the snapshot: variant: inverted, size: xl, icon: true, children: false 1`] = `
<div>
  <button
    class="b11jtqo5 bdi40cx b1vby7g1"
    data-testid="button"
  >
    <svg
      class="c108jgnp bkz0lf"
      fill="none"
      height="24"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M12 1a3 3 0 0 0-3 3v8a3 3 0 0 0 6 0V4a3 3 0 0 0-3-3z"
      />
      <path
        d="M19 10v2a7 7 0 0 1-14 0v-2"
      />
      <line
        x1="12"
        x2="12"
        y1="19"
        y2="23"
      />
      <line
        x1="8"
        x2="16"
        y1="23"
        y2="23"
      />
    </svg>
    <span
      class="slvi054 b276tsp"
    />
  </button>
</div>
`;

exports[`<Button> matches the snapshot: variant: inverted, size: xl, icon: true, children: true 1`] = `
<div>
  <button
    class="b11jtqo5 bdi40cx b1vby7g1"
    data-testid="button"
  >
    <svg
      class="c108jgnp bkz0lf c75i3tf"
      fill="none"
      height="24"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M12 1a3 3 0 0 0-3 3v8a3 3 0 0 0 6 0V4a3 3 0 0 0-3-3z"
      />
      <path
        d="M19 10v2a7 7 0 0 1-14 0v-2"
      />
      <line
        x1="12"
        x2="12"
        y1="19"
        y2="23"
      />
      <line
        x1="8"
        x2="16"
        y1="23"
        y2="23"
      />
    </svg>
    <span
      class="slvi054 b276tsp"
    >
      Lorem Ipsum
    </span>
  </button>
</div>
`;
